crates/blob-store/src/implementors/aws_s3.rs

```diff
@@ -0,0 +1,254 @@
+        Ok(self.bucket.name().unwrap_or_default().to_string())
+    }
+    async fn info(&self) -> Result<ContainerMetadata> {
+        Ok(ContainerMetadata::from(&self.bucket))
+    }
+    async fn list_objects(&self) -> Result<Vec<ObjectNameResult>> {
+        let resp = self
+            .client
+            .list_objects_v2()
+            .bucket(self.name().await?)
+            .send()
+            .await?;
+        info!("{}", "received list objects response");
+        let res = resp
+            .contents()
+            .unwrap_or_default()
+            .iter()
+            .map(|object| object.key().unwrap_or_default().to_string())
+            .collect();
+        Ok(res)
+    }
+    async fn delete_object(&self, name: ObjectNameParam<'_>) -> Result<()> {
+        let _ = self
+            .client
+            .delete_object()
+            .bucket(self.name().await?)
+            .key(name)
+            .send()
+            .await?;
+        info!("{}", format!("object {name} deleted"));
+        Ok(())
+    }
+    async fn delete_objects(&self, names: Vec<ObjectNameParam<'_>>) -> Result<()> {
+        let mut delete_objects: Vec<ObjectIdentifier> = vec![];
+        for name in names {
+            let obj_id = ObjectIdentifier::builder()
+                .set_key(Some(name.to_owned()))
+                .build();
+            delete_objects.push(obj_id);
+        }
+        let delete = Delete::builder().set_objects(Some(delete_objects)).build();
+        let _ = self
+            .client
+            .delete_objects()
+            .bucket(self.name().await?)
+            .delete(delete)
+            .send()
+            .await?;
+        info!("{}", "objects deleted");
+        Ok(())
+    }
+    async fn has_object(&self, name: ObjectNameParam<'_>) -> Result<bool> {
+        let res = self
+            .client
+            .get_object()
+            .bucket(self.name().await?)
+            .key(name)
+            .send()
+            .await;
+        let mut key_exists: bool = true;
+        if let Err(err) = res {
+            match err.into_service_error() {
+                GetObjectError {
+                    kind: GetObjectErrorKind::NoSuchKey(_),
+                    ..
+                } => key_exists = false,
+                err => return Err(err.into()),
+            }
+        }
+        Ok(key_exists)
+    }
+    async fn object_info(&self, name: ObjectNameParam<'_>) -> Result<ObjectMetadata> {
+        let container = self.name().await?;
+        let metadata = self
+            .client
+            .get_object_attributes()
+            .bucket(container.clone())
+            .key(name)
+            .object_attributes(aws_sdk_s3::model::ObjectAttributes::ObjectSize)
+            .send()
+            .await?;
+        let res = ObjectMetadata {
+            name: name.to_owned(),
+            container,
+            created_at: metadata.last_modified().unwrap().as_secs_f64() as u64, // TODO: fix me
```

Comment:
It looks like it is based on the output from the cli: https://docs.aws.amazon.com/cli/latest/reference/s3api/get-object-attributes.html

It wasn't clear to me from the documentation, but it seems reasonable that it would be.

Reply:
Yes I believe so. 

What I am not sure in this impelemntation is whether or not `f64` can be just converted to `u64`. Seems like a lot of information got lost in the transition.